autofs-5.1.0 - force disable browse mode for amd format maps

From: Ian Kent <ikent@redhat.com>

As in am-utils amd format maps using autofs file system don't support
the browse option and in autofs they don't work properly if it is used.

This might be implemented later but for now forcefully disable it.
---
 CHANGELOG              |    1 +
 lib/master_parse.y     |    2 ++
 modules/mount_autofs.c |    2 ++
 3 files changed, 5 insertions(+)

diff --git a/CHANGELOG b/CHANGELOG
index 31a804d..f291095 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -21,6 +21,7 @@
 - fix copy and paste error in dup_defaults_entry().
 - fix leak in parse_mount().
 - add mutex call return check in defaults.c.
+- force disable browse mode for amd format maps.
 
 04/06/2014 autofs-5.1.0
 =======================
diff --git a/lib/master_parse.y b/lib/master_parse.y
index e31023d..9da78fc 100644
--- a/lib/master_parse.y
+++ b/lib/master_parse.y
@@ -810,6 +810,8 @@ int master_parse_entry(const char *buffer, unsigned int default_timeout, unsigne
 			logopt = LOGOPT_DEBUG;
 		else if (loglevel <= LOG_INFO && loglevel > LOG_ERR)
 			logopt = LOGOPT_VERBOSE;
+		/* amd mounts don't support browse mode */
+		ghost = 0;
 	}
 
 
diff --git a/modules/mount_autofs.c b/modules/mount_autofs.c
index c7cba3a..4846e7f 100644
--- a/modules/mount_autofs.c
+++ b/modules/mount_autofs.c
@@ -261,6 +261,8 @@ int mount_mount(struct autofs_point *ap, const char *root, const char *name,
 			nap->pref = am_entry->pref;
 			am_entry->pref = NULL;
 		}
+		/* amd mounts don't support browse mode */
+		nap->flags &= ~MOUNT_FLAG_GHOST;
 	}
 
 	if (handle_mounts_startup_cond_init(&suc)) {
